### ISSUE TITLE
DII / DIO Protocol 

### ISSUE THREAD

**Issue by v-hofmann** [cortex-lab/neuropixels#32]
Hi everyone,

we have started using Neuropixels for recording in electric fish (Apteronotus leptorynchus) a while ago. Now that the technical and analytical parts are more or less established we are focussing on getting the histology going for recovering the penetration tracts of the electrodes - the problem is that no one currently being in our lab has substantial experience with histology of flourescent dyes. Thus for the past couple of months, even though we have tried a lot, we didnt have any luck in recovering any flourescent signatures in our slides. 

here a quick description of what we tried: 
- we have used DII and DIO (1% sollution from crystals in IPA) as well as the already dissolved sollutions, put a drop on a pipette and covered the electrodes with it. 
- duration of the recording session were about 3-5 hours
- post electrode removal fish were perfused transcardially (PBS -> 4% PFA -> 4% PFA +30% SUC)
- we have had slides (20 mum) both cut on the cryostat or using paraffin embedding (then no 30% SUC) 
- we have used no background stains and slides we not embedded (i.e. to just see the signature first) but would use Cresy violet of DAPI background stain in the future if possible 
- for DII we are using green excitation light. The whole slide lights up in red in contrast to what i see [here](https://github.com/cortex-lab/neuropixels/issues/21) (but the fact that this is blue is because of a DAPI background stain i guess?) 

I am also attaching some pics of our best slides (in these ones exclusively we see a systematic electrode tract and a somewhat flourescent siguature around it - but far from what we would expect based on the published or posted examples... 

So does anyone have tips on what we should adjust in order to have better luck? 

Is it the coating of the electrodes? (they were visibly coated in the experiment shown in the pics) - is there anything we should think of in addition to [this](https://github.com/cortex-lab/neuropixels/wiki/Identifying_tracks_in_histology)

Is it the post-processing that creates the background flourescense so that we dont see our red signal?  Does anyone have a protocol for the post processing of the brains she/he could share? or some advise on what is best to do? (cryo vs vibratome vs paraffin? What counterstain work with the dyes best?) 

looking forward to any advise... 

best
Volker

![01_full](https://user-images.githubusercontent.com/51334456/73753107-1e440d00-4730-11ea-8135-82e3698ad3cc.jpg)
![01_zoom](https://user-images.githubusercontent.com/51334456/73753108-1edca380-4730-11ea-9cf6-1868ba18a968.jpg)
![02_zoom](https://user-images.githubusercontent.com/51334456/73753110-1edca380-4730-11ea-8271-2dcda059894e.jpg)

).

**Comment by petersaj** [comment]
We don't do post-processing or anything fancy to get those signals, they're
usually that clear as-is. What's your prep - does the probe go through
liquid or agar on it's way towards the fish, is it possible the dye is
coming off during the approach? It might be worth trying to get a positive
control by fixing a brain, puncturing it with a needle or something similar
coated in dye, then slicing?

**Comment by v-hofmann** [comment]
Thanks for the suggestion - i have actually tried something like that without luck (i had put some crystals on the brain surface and stabbed the brain with a needle several times). I was under the impression that the time the needle with the dye was in the already perfused brain was too short - I think i will give it another try with coating the needles similar what you probes.

regarding the other questions: the experiments are in vivo, i.e. the electrode goes through a little saline bath (not seeing any dye coming of there). When it goes into the brain, i have seen some dye accumulating at the surface (i.e. i am sure some went off) but i would think that this is what usually happens (i.e. after extracting the brain i saw a pink dot on the surface). 

Can you tell me what embedding/cutting procedure you use? Are you cutting on a Cryostat? Any EtOH or Xylol involved in the embedding (which is supposedly bad for the flourescent signal)? Do you have problems with photobleaching of the dye, i.e. are you very careful not to have the slides exposed to light? Do you use a background/counterstain? Does your raw slides look similar to what i have posted (light up in orange)? 

everyone keeps telling us its easy - makes it only more frustrating to not be able to get it right :)

**Comment by petersaj** [comment]
Ah yeah going through a little saline sounds unlikely to have it all come
off. For coating probes - it sounds like you're doing what we do, we think
the most important thing is that the DiI is allowed to evaporate onto the
probe, so you've got to drag the drop slowly up and down the probe until
the drop evaporates, or more recently we've been dipping the probe quickly
in dye and letting it dry for ~2-3 seconds, then repeat ~6x. For what it's
worth though I've never seen a pink dot on the surface after recording...

Our slices are cut on a cryostat, we use Mowiol as a mounting medium, and
haven't had any issues with photobleaching (not particularly careful with
light exposure). We usually stain with DAPI (and in my case I've also got
GCaMP in the green), but the red signal usually has contrast like the
github image rather than having a lot of background/autofluorescence like
in your images

**Comment by v-hofmann** [comment]
Its well possible that it is our coating procedure. In the 1st experiments i vertically dipped the probe in the epi with the sollution 10-20 times for ~ 1 minute and then pulled it out for 1-2 minutes to have the IPA evaporated. More recently i oriented the probe horizontally and ran a drop with a pipette along it (this was when i saw it notably being pink). 

the Autoflourescence is a big concern... - are you using 4% PFA = 30% SUC for perfusion too? I didn't hear of PFA causing that - mostly people tell be to avoid Gultardialdehyde for that reason...

**Comment by petersaj** [comment]
We use 4% PFA, but I forgot we've been using a vibratome instead of a
cryostat, so no sucrose, no idea if that affects the autofluorescence?

**Comment by nsteinme** [comment]
Have you tried an even simpler positive control of just putting some dii on
a slide under your microscope and making sure you can see it fluoresce?
I.e. maybe something is confused about the color channels somehow?

It might just be the case that your species has stronger/different
autofluorescence than mammals, in which case you might try to find a scope
with many different color channels and see whether you can find one color
that has autofluorescence alone, and one that has autofluorescence+dii, and
see whether a subtraction of the two can yield a better signal.

You could also try the CM-DiI which is modified to bind to some other
things in the tissue and might stick around better in case the dii is
somehow getting washed away in some step of your procedure. But I don't
have any particular suggestions about procedural elements to change.

**Comment by v-hofmann** [comment]
Yes i have tried the dye on the slide - looks somewhat like the lid up brain (but the manuals seem to suggest that the signal gets amplified once the dye is taken into the membranes...). 
 
ill try to find another microscope and double check - i.e. our excitation light is greenish and i see organeish light in the ocular (which somewhat matches [this](https://www.thermofisher.com/order/catalog/product/V22885?SID=srch-hj-V22885#/V22885?SID=srch-hj-V22885) but noone is able to tell me the exact wavelengths of the filters in the scope we have (which was bought back in the days...). 

thanks for the feedback everyone!
